drivers/spi/spi_esp32_spim.c

```diff
@@ -291,7 +286,8 @@ static int IRAM_ATTR spi_esp32_configure(const struct device *dev,
 	}
 
 	/* enables SPI peripheral */
-	if (clock_control_on(cfg->clock_dev, cfg->clock_subsys)) {
+	res = clock_control_on(cfg->clock_dev, cfg->clock_subsys);
+	if (res != 0 && res != -EALREADY) {
```

Comment:
I see a lot of code formatting in this PR. Could you only modify the code where it fixes the issue? I guess that it this if check?

Reply:
Yeah, my mistake, sorry about that.